lib/rich-text.rb

```diff
@@ -15,17 +15,38 @@ def self.configure
 
 RichText.configure do |c|
   c.safe_mode = true
-  c.html_default_block_tag = 'p'
-  c.html_block_tags = {
-    firstheader: 'h1',
-    secondheader: 'h2',
-    thirdheader: 'h3',
-    list: ->(content, value) { %(<ol><li>#{value}</li></ol>) },
-    bullet: ->(content, value) { %(<ul><li>#{value}</li></ul>) }
-  }
-  c.html_inline_tags = {
-    bold: 'strong',
-    italic: 'em',
-    link: ->(content, value) { %(<a href="#{value}">#{content}</a>) }
+
+  c.html_default_block_format = :paragraph
+
+  c.html_formats = {
+    bold:           { type: :inline, tag: 'strong' },
+    br:             { type: :inline, tag: 'br' },
+    bullet:         { type: :block,  tag: 'li', parent: 'ul' },
+    div:            { type: :block,  tag: 'div' },
+    fifthheader:    { type: :block,  tag: 'h5' },
+    figure:         { type: :block,  tag: 'figure' },
+    firstheader:    { type: :block,  tag: 'h1' },
+    fourthheader:   { type: :block,  tag: 'h4' },
+    hr:             { type: :inline, tag: 'hr', block_format: false },
+    id:             { type: :block,  apply: ->(el, op){ el[:id] = op.attributes[:id] } },
+    image:          { type: :inline, tag: 'img', block_format: :figure, apply: ->(el, op){ el[:src] = op.value.dig(:image, :src) } },
+    ins:            { type: :inline, tag: 'ins' },
+    italic:         { type: :inline, tag: 'em' },
+    link:           { type: :inline, tag: 'a', apply: ->(el, op){ el[:href] = op.attributes[:link] } },
+    list:           { type: :block,  tag: 'li', parent: 'ol' },
+    paragraph:      { type: :block,  tag: 'p' },
```

Comment:
I guess I took the approach of: this is a definition of stuff that can show up in your rendered output. If it intersects with your schema, use a different name for the helper format. Either way though, I'd want the renderer to produce artifacts explicitly as I'd spec'd them.

Reply:
I've addresses this so that block formats intended as structural helpers don't get definitions. So, elements found in the delta will not respond to those helper formats.